File: weechat-android/src/main/java/com/ubergeek42/WeechatAndroid/relay/BufferEye.java
/**
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 */

package com.ubergeek42.WeechatAndroid.relay;

import android.support.annotation.UiThread;

public interface BufferEye {

    // 1 line added on bottom
    void onLineAdded();

    // all lines should be re-rendered due to font size change and such
    @UiThread void onGlobalPreferencesChanged(boolean numberChanged);

    // server sent us all lines
    void onLinesListed();

    void onPropertiesChanged();

    void onBufferClosed();
}

File: app/src/main/java/com/ubergeek42/WeechatAndroid/relay/BufferEye.kt
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
package com.ubergeek42.WeechatAndroid.relay

import androidx.annotation.MainThread
import androidx.annotation.WorkerThread

interface BufferEye {
    // server sent us all lines
    @WorkerThread fun onLinesListed()

    // 1 line added on bottom
    @WorkerThread fun onLineAdded()

    // indicates changed title
    @WorkerThread fun onTitleChanged()

    // buffer was closed in weechat
    @WorkerThread fun onBufferClosed()

    // all lines should be re-rendered due to font size change and such
    @MainThread fun onGlobalPreferencesChanged(numberChanged: Boolean)
}